corehq/pillows/ledger.py

```diff
@@ -57,6 +62,27 @@ def _get_daily_consumption_for_ledger(ledger):
     return daily_consumption
 
 
+def _update_ledger_section_entry_combinations(ledger):
+    current_combos = _get_ledger_section_combinations(ledger['domain'])
+    if (ledger['section_id'], ledger['entry_id']) in current_combos:
+        return
+
+    # clear the lru_cache so that next time a ledger is saved, we get the combinations
+    _get_ledger_section_combinations.cache_clear()
```

Comment:
shouldn't this happen after the new entry is added

Reply:
Technically since I'm using a local LRU cache then I think this was safe before, but clearing cache later does make sense if there was a chance for parallel queries https://github.com/dimagi/commcare-hq/pull/25304/commits/f9184cf162e4005d5ca3ff975bbac0250892b994